File: app/src/main/java/com/loopeer/codereader/api/BaseListResponse.java
package com.loopeer.codereader.api;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class BaseListResponse<T> {

    @SerializedName("total_count")
    public int totalCount;
    @SerializedName("incomplete_results")
    public boolean incompleteResults;
    @SerializedName("items")
    public List<T> items;

}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/api/BaseListResponse.kt
package com.loopeer.codereaderkt.api

import com.google.gson.annotations.SerializedName


class BaseListResponse<T> {

    @SerializedName("total_count")
    var totalCount: Int = 0
    @SerializedName("incomplete_results")
    var incompleteResults: Boolean = false
    @SerializedName("items")
    var items: List<T>? = null

}